scalableminds/webknossos
# Allow task type summary to identify task type when creating tasks in bulk

### URL of deployed dev instance (used for testing):
- https://tasktypebysummary.webknossos.xyz

### Steps to test (end-to-end):
 - Create tasks in bulk, try by task type id and by summary. example:
```
ROI2017_wkw, sampleTaskType, sampleExp, 1, 512,512,512, 0, 0, 0, 1, 500, 500, 500, 100, 100, 100, sampleProject
ROI2017_wkw, sampleTaskType, sampleExp, 1, 512,512,512, 0, 0, 0, 1, 500, 500, 500, 100, 100, 100, sampleProject
ROI2017_wkw, sampleTaskType, sampleExp, 1, 512,512,512, 0, 0, 0, 1, 500, 500, 500, 100, 100, 100, sampleProject
```

Note that by summary does not work if the task type summary contains commas, since that breaks the frontend csv parsing (quote escaping is not implemented there).  However, we checked two large wk instances and nobody had task types with commas anyway

### Steps to test (backend only):
- Create a task via the POST /api/tasks route using a taskTypeSummary in the field taskTypeIdOrSummary rather than a taskTypeId
- Create a task via the POST /api/tasks route using a taskTypeId to see that it still works

### Issues:
- fixes #4990

------
